app/src/main/kotlin/com/intive/picover/main/viewmodel/MainViewModel.kt

```diff
@@ -1,17 +1,26 @@
 package com.intive.picover.main.viewmodel
 
-import androidx.lifecycle.ViewModel
+import androidx.compose.material3.SnackbarHostState
+import cafe.adriel.voyager.core.model.ScreenModel
 import com.intive.picover.auth.repository.AuthRepository
+import com.intive.picover.main.navigation.launcher.Launcher
+import com.intive.picover.main.navigation.launcher.LauncherEvent
 import com.intive.picover.main.viewmodel.state.MainState
-import dagger.hilt.android.lifecycle.HiltViewModel
 import javax.inject.Inject
 import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.onEach
 
-@HiltViewModel
 class MainViewModel @Inject constructor(
 	authRepository: AuthRepository,
-) : ViewModel() {
+	launcher: Launcher,
+	val snackbarHostState: SnackbarHostState,
```

Comment:
Is it needed?

Reply:
It's accessed in MainScreen to set up snackbar host

Reply:
Right, now I see